test/albumController.test.js

```diff
@@ -173,24 +155,151 @@ describe('AlbumController API Tests', () => {
+    };
+
+    const response = await request(app)
+      .post('/api/v1/albums')
+      .set('Cookie', signedCookie)
+      .send(newAlbumData);
+
+    expect(response.status).toBe(StatusCodes.CREATED);
+    expect(response.body).toHaveProperty('album');
+  });
+
+  it('should create a new album - Error Case (Invalid Data)', async () => {
+    const adminCredentials = {
+      email: 'admin@admin.com',
+      password: 'adminpassword',
+    };
+
+    const signedCookie = await loginAndReturnCookie(adminCredentials);
+
+    const invalidAlbumData = {
+      artistName: 'New Artist',
+      price: 12.99,
+      spotifyUrl: 'https://api.spotify.com/v1/albums/new123',
+    };
+
+    const response = await request(app)
+      .post('/api/v1/albums')
+      .set('Cookie', signedCookie)
+      .send(invalidAlbumData);
+
+    expect(response.status).toBe(StatusCodes.BAD_REQUEST);
+  });
+
+  it('should update an existing album - Success Case', async () => {
+    const adminCredentials = {
+      email: 'admin@admin.com',
+      password: 'adminpassword',
+    };
+
+    const signedCookie = await loginAndReturnCookie(adminCredentials);
+
+    const existingAlbum = await Album.findOne({});
+    const updatedData = {
+      albumName: 'Updated Album Name',
+    };
+
+    const response = await request(app)
+      .patch(`/api/v1/albums/${existingAlbum._id}`)
+      .set('Cookie', signedCookie)
+      .send(updatedData);
+
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
+    const updatedAlbum = await Album.findById(existingAlbum._id);
+
+    expect(updatedAlbum).not.toBeNull();
+    expect(updatedAlbum.albumName).toBe(updatedData.albumName);
+  });
+
+  it('should update an existing album - Error Case (Not Found)', async () => {
+    const adminCredentials = {
+      email: 'admin@admin.com',
+      password: 'adminpassword',
+    };
+
+    const signedCookie = await loginAndReturnCookie(adminCredentials);
+
+    const invalidAlbumId = 'invalidAlbumId';
+    const updatedData = {
+      albumName: 'Updated Album Name',
+    };
+
+    const response = await request(app)
+      .patch(`/api/v1/albums/${invalidAlbumId}`)
+      .set('Cookie', signedCookie)
+      .send(updatedData);
+
+    expect(response.status).toBe(StatusCodes.NOT_FOUND);
+  });
+
+  it('should update album prices - Success Case', async () => {
+    const adminCredentials = {
+      email: 'admin@admin.com',
+      password: 'adminpassword',
+    };
+
+    const signedCookie = await loginAndReturnCookie(adminCredentials);
+
+    const existingAlbum = await Album.findOne({});
+    const updatedData = {
+      id: existingAlbum._id,
+      price: 14.99,
+    };
+
+    const response = await request(app)
+      .patch(`/api/v1/albums/${existingAlbum._id}`)
+      .set('Cookie', signedCookie)
+      .send(updatedData);
+
+    expect(response.status).toBe(StatusCodes.OK);
+    expect(response.body).toHaveProperty('album');
```

Comment:
Liek you've done in the `should update an existing album - Success Case` test , it would be good to also have a check here to verify that the album in the db was actually successfully updated.

Reply:
Done , commit 1b7dec473b617cdcff936d78a14387142cde8cac